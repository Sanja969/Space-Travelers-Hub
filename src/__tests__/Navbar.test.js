import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../components/Navbar';

describe('Test Navbar render', () => {
  it('Test Navbar snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <NavBar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
