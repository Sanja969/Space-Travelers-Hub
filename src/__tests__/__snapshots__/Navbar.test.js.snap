// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar matches the snapshot 1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Test Navbar render  1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Test Navbar render Test  1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Test Navbar render Test B 1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Test Navbar render Test Navbar snapshot 1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Test Navbar render matches the snapshot 1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`test the Quote components matches the snapshot 1`] = `
<nav
  className="container-fluid navbar navbar-expand-md navbar-light bg-white fixed-top border-bottom"
>
  <div
    className="container-fluid d-flex align-items-center mx-0 px-0"
  >
    <div
      className="navbar-brand fw-bold fs-3 ms-0 ps-0"
    >
      <img
        alt="logo"
        src="logo.png"
        width="30"
      />
      <span
        className="ps-3 fw-bold"
      >
        The Space Travelers' Hub
      </span>
    </div>
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse text-end justify-content-end text-center"
      id="navbarTogglerDemo01"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/rockets"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Rockets
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/missions"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              Missions
            </p>
          </a>
        </li>
        <li
          className="link"
        >
          <a
            className="nav-item text-decoration-none"
            href="/myProfile"
            onClick={[Function]}
          >
            <p
              className="nav-link pt-4 text-primary"
            >
              My Profile
            </p>
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
